KenticoCloud.Delivery/DeliveryClient.cs

```diff
@@ -521,5 +521,17 @@ public DeliveryClient(string projectId, string previewApiKey) : this(new Deliver
 
             throw new DeliveryException(response.StatusCode, await response.Content.ReadAsStringAsync());
         }
+
+        private Uri GetUri(string url)
+        {
+            try
+            {
+                return new Uri(url);
+            }
+            catch (UriFormatException ex)
+            {
+                throw new Exception("The request url is too long. Split your query into multiple calls.", ex);
```

Comment:
I'm kinda missing the point here. Why to wrap the original exception? It already contains the information about the URL being too long. Or is it just to add the extra hint about splitting the query into multiple calls?

Reply:
Yeah, I wanted to write some suggestion because the user of the SDK should know how to fix this problem.

Reply:
right...on the other hand - you don't know if the UriFormatException will be caused by the length of the query. so your message could be completely off in some cases

Reply:
I would also not throw general `Exception`, but rather use something more specific